src/RepoDriver.sol

```diff
@@ -0,0 +1,508 @@
+    }
+
+    /// @notice Requests the AnyApi operator to update the repository owner.
+    /// See `requestUpdateRepoOwner` for more details.
+    /// @param forge The forge where the repository is stored.
+    /// @param name The name of the repository.
+    /// @param fee The fee in Link to pay for the request.
+    function _requestUpdateRepoOwner(Forge forge, bytes memory name, uint256 fee) internal {
+        RepoDriverAnyApiStorage storage storageRef = _repoDriverAnyApiStorage();
+        address operator = address(storageRef.operator);
+        require(operator != address(0), "Operator address not set");
+        uint256 nonce = storageRef.nonce++;
+        bytes32 requestId = keccak256(abi.encodePacked(this, nonce));
+        uint224 repoId = calcRepoId(forge, name);
+        storageRef.requestedUpdates[requestId] = repoId;
+        bytes memory payload = _requestPayload(forge, name);
+        bytes memory callData = abi.encodeWithSelector(
+            OperatorInterface.operatorRequest.selector,
+            address(0), // ignored, will be replaced in the operator with this contract's address
+            0, // ignored, will be replaced in the operator with the fee
+            storageRef.jobId,
+            this.updateRepoOwnerByAnyApi.selector,
+            nonce,
+            2, // data version
+            payload
+        );
+        require(linkToken.transferAndCall(operator, fee, callData), "Transfer and call failed");
+        // slither-disable-next-line reentrancy-events
+        emit RepoOwnerUpdateRequested(repoId, forge, name);
+    }
+
+    /// @notice Builds the AnyApi generic `bytes` fetching request payload.
+    /// It instructs the operator to fetch the current owner of the repository.
+    /// @param forge The forge where the repository is stored.
+    /// @param name The name of the repository.
+    /// @return payload The AnyApi request payload.
+    function _requestPayload(Forge forge, bytes memory name)
+        internal
+        view
+        returns (bytes memory payload)
+    {
+        // slither-disable-next-line uninitialized-local
+        BufferChainlink.buffer memory buffer;
+        buffer = BufferChainlink.init(buffer, 256);
+        buffer.encodeString("get");
+        buffer.encodeString(_requestUrl(forge, name));
+        buffer.encodeString("path");
+        buffer.encodeString(_requestPath());
+        return buffer.buf;
+    }
+
+    /// @notice Builds the URL for fetch the `FUNDING.json` file for the given repository.
+    /// @param forge The forge where the repository is stored.
+    /// @param name The name of the repository.
+    /// @return url The built URL.
+    function _requestUrl(Forge forge, bytes memory name)
+        internal
+        pure
+        returns (string memory url)
+    {
+        if (forge == Forge.GitHub) {
+            return string.concat(
+                "https://raw.githubusercontent.com/", string(name), "/HEAD/FUNDING.json"
```

Comment:
Ah nice, I didn't know that the `HEAD` will just always result in the default branch. This simplifies it a lot.

Reply:
Yes, it does, I didn't know this either and was so happy to discover this :raised_hands: